adapter/internal/oasparser/model/http_route.go

```diff
@@ -289,6 +307,61 @@ func concatAuthScheme(scheme *dpv1alpha1.Authentication) *dpv1alpha1.Authenticat
 	return &finalAuth
 }
 
+// concatAPIPolicies concatinate override and default authentication rules to a one authentication override rule
+// folowing the hierarchy described in the https://gateway-api.sigs.k8s.io/references/policy-attachment/#hierarchy
+// Following code would follow below logic.
+// | API override policies | Resource override policies | Resource default policies | API default policies |
+// |            1          |              1             |              0            |           0          | API override policies + Resource override policies
+// |            1          |              0             |              1            |           0          | API override policies
+// |            0          |              1             |              0            |           1          | Resource override policies
+// |            0          |              0             |              1            |           1          | Resource default policies
+func concatAPIPolicies(schemeUp *dpv1alpha1.APIPolicy, schemeDown *dpv1alpha1.APIPolicy) *dpv1alpha1.APIPolicy {
+	if schemeUp == nil && schemeDown == nil {
+		return nil
+	} else if schemeUp == nil {
+		return schemeDown
+	} else if schemeDown == nil {
+		return schemeUp
+	}
+
+	finalAPIPolicy := dpv1alpha1.APIPolicy{}
+
+	// api level override policies - must apply
+	finalAPIPolicy.Spec.Override.RequestQueryModifier = schemeUp.Spec.Override.RequestQueryModifier
```

Comment:
Shouldn't we use the order like this?
Starting with an empty `RequestQueryModifier` struct and fill it in right to left order.
```
API level override > Resource level override > Resource level default > API level default 
```
(**default are applying from bottom to up, override are applying from top to bottom**)

Do we need to combined properties like you have done using append. I think override should replace instead (considering the order as above). I'm referring to the diagram in the [spec](https://gateway-api.sigs.k8s.io/references/policy-attachment/#hierarchy)   

Seems this default/override logic will repeat for the Policies we introduce in future. We will have to add similar functions again. Is it possible to make this a generic function like below.

```
type PolicyType struct {
	Default  any
	Override any
}

func resolveFinalPolicy[PolicyType, V any](schemeUp PolicyType, schemeDown PolicyType) V {
	var finalPolicy V
	// logic to find the actual properties applied to resource
	return finalPolicy
}
```

Reply:
+1. I too would love to add a common util like the above. but how can we determine at which level the policies are added? these are structs and vary according to the crd type, i.e. it's not like a simple nil check. Please let me know if you have a solution to determe which level is given and not given.

Reply:
This [reflect package](https://pkg.go.dev/reflect) might help to iterate with arbitrary types of policies. shall we create an issue to take a look at it again since we are moving to other tasks now?

Reply:
this will be addressed with a new issue https://github.com/wso2/apk/issues/691